### PR TITLE
Migrate SunPy to support Python 3 and 2 simultaneously

### PR DESCRIPTION
**Pull request by embray**
This PR is to get things started on fixing #371 and other issues labeled Python-3.x.  This PR might not attempt to do *everything*, but it should provide a starting basis that can be built on (perhaps as PRs to this PR sent to my SunPy fork).

Here's a *very* rough overview of things that will need to be dealt with:

* [x] Adds a local copy of the `six` module to aid with migration.
* [x] Run python-modernize over all submodules as a starting point.
* [ ] Review changes by python-modernize and clean up ones that are incorrect or unnecessary (in particular sometimes it will replace dict iterations in cases where it really isn't important).
* [ ] Fix all broken imports (using `six` as much as possible)
* [ ] Fix any syntax errors
* [ ] Fix any dict iterations to use the `itervalues` and `iteritems` from `six`, and similar where necessary
* [ ] Deal with string-related issues.
* [ ] Deal with networking-related issues (urllib, etc.)
* [ ] Make sure all relevant docs mention the suds-jurko fork of suds

### CODE REVIEW

**Comment by Cadair** [comment]
@embray is there an advantage to including six ourselves rather than using the astropy version?

**Comment by embray** [comment]
Yes--otherwise even from within the source checkout `import sunpy` completely fails if Astropy isn't already available too.  Even if you have Astropy as a dependency it's advantageous to include your own copy of six.

**Comment by embray** [comment]
Blindly ran python-modernize over the sunpy package, one 2nd-level submodule at a time for easier digestion.

**Comment by embray** [comment]
Rebased on master.  `./setup.py build` works now, at least, on python3.  That doesn't guarantee much else though.

**Comment by embray** [comment]
That's all I got time for right now, but it's a start.  Will have to see about adding a Python 3 build for AppVeyor.  In the meantime it will be interesting just to see what this broke on Python 2 (no doubt something--`unicode_literals` alone tends to break things in unexpected ways).

**Comment by Cadair** [comment]
Don't know why Travis hasn't rerun on your rebased build here...

**Comment by embray** [comment]
I dunno ?

**Comment by Cadair** [comment]
I opened a PR to your branch that enables Python 3 testing on Travis, also it fixes the `conftest.py` to be 3.x compat.

It seems that there is a problem with the ana c extension building under Python 2.

**Comment by embray** [comment]
There--merging your PR seems to have given Travis a kick.

File: README.md [hunk 1]
```diff
@@ -8,12 +8,12 @@ Installation
 
 To begin, install the following requirements:
 
- * [Python](http://www.python.org) (2.7+)
+ * [Python](http://www.python.org) (2.7)
```
Comment (embray) [inline README.md]: Maybe "(2.7, 3.3+)"

**Comment by cdeil** [comment]
Just wanted to try out something in Sunpy and got this:
```
$ python setup.py install --user
Traceback (most recent call last):
  File "setup.py", line 32, in <module>
    from sunpy.tests.setup_command import SunPyTest
  File "/Users/deil/code/sunpy/sunpy/__init__.py", line 24, in <module>
    from sunpy.util.config import load_config, print_config
  File "/Users/deil/code/sunpy/sunpy/util/__init__.py", line 4, in <module>
    from sunpy.util.util import *
  File "/Users/deil/code/sunpy/sunpy/util/util.py", line 6, in <module>
    from itertools import izip, imap, count
ImportError: cannot import name 'izip'
```
Thanks @embray for working on Python 3 support!

**Comment by Cadair** [comment]
@cdeil Thanks for the comment, it's useful to have some idea that people want to use it under Python 3. @astrofrog also pointed this out to me the other day.

Any help is welcome on this.

**Comment by Cadair** [comment]
@embray @ehsteve @nabobalis 

Shall we move this into a branch on the main repository, which would make it easier for all of us to rebase it / have multiple people working on it, without putting too much burden on @embray ?

**Comment by nabobalis** [comment]
If we do plan to do this, that sounds like a decent idea.

**Comment by wafels** [comment]
I agree

**Comment by astrofrog** [comment]
Just for information when doing this in Astropy we actually did it one sub-package at a time (i.e. over many PRs). When using astropy-helpers and the setup_package.py files, you can disable 2to3 on a sub-package level, so that's what we did.

**Comment by astrofrog** [comment]
Another way to deal with this in smaller chunks is to start off by just adding the ``__future__`` imports and fixing any resulting failures on Python 2, then merging once Python 2 works. Then you can fix any issues on Python 3 in a separate PR.

**Comment by Cadair** [comment]
@astrofrog getting this to pass python 2 then merging it would be a good start.

**Comment by embray** [comment]
For anyone who's been trying to use this branch, the work is incomplete.

**Comment by Cadair** [comment]
closing this in favor of doing it bit by bit.

**Pull request closed without merging**